### PR TITLE
s2i OHC

### PR DESCRIPTION
**Pull request by praniredhat**
Please review @aksgupta14  @anujsi @rh-kgarg @vrathee

### CODE REVIEW

File: bump_version.sh [hunk 1]
```diff
@@ -9,5 +9,5 @@ versiony --version=$1 --to=package.json,bower.json &&
 npm install &&
 git commit -am "Publish version $1" &&
 git tag $1 &&
```
Comment (praniredhat) [inline bump_version.sh]: Remove this change

File: app/server.js [hunk 2]
```diff
@@ -19,7 +18,7 @@ function setCustomCacheControl(res, path) {
         res.setHeader('Cache-Control', 'public, max-age=600');
     }
 }
-
+console.log('dir and filename:', __dirname, __filename);
```
Comment (vrathee) [inline app/server.js]: @praniredhat  Do we need this console log?

File: app/server.js [hunk 3]
```diff
@@ -62,8 +61,8 @@ try {
     const key = fs.readFileSync('/etc/ssl/server.key');
 
     const options = {
-        cert,
-        key
+        cert: cert,
+        key: key
```
Comment (vrathee) [inline app/server.js]: we can keep them as 
```const options = {
	        cert,
	        key
}```

File: package.json [hunk 4]
```diff
@@ -88,10 +96,10 @@
     "test": "./node_modules/.bin/karma start",
     "build": "rimraf dist && nice webpack --config webpack-production.config.js --bail --progress --profile",
     "server-inline": "webpack-dev-server --config webpack-dev-server.config.js --history-api-fallback --inline --progress --port 9000",
-    "server": "webpack-dev-server --config webpack-dev-server.config.js --history-api-fallback --progress --port 9000",
-    "server-prod": "webpack-dev-server --config webpack-production.config.js --history-api-fallback --inline --progress --port 9000",
+    "server": "cp hooks/pre-commit .git/hooks/ && webpack-dev-server --config webpack-dev-server.config.js --history-api-fallback --progress --port 9000",
+    "server-prod": "webpack-dev-server --config webpack-production.config.js --history-api-fallback --progress --port 9000",
```
Comment (vrathee) [inline package.json]: Are the scripts foe server, server-prod, server-inline accurate?

File: webpack-production.config.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
-module.exports = require("./webpack.config")({
+module.exports = [require("./webpack.config")({
```
Comment (vrathee) [inline webpack-production.config.js]: Why do we need to return an Array now?

File: app/server.js (hunk 2 above)
Comment (praniredhat) [reply app/server.js]: done

File: webpack-production.config.js (hunk 5 above)
Comment (praniredhat) [reply webpack-production.config.js]: this has to be array with 2 items, I added one more item, that time i had to remove it temporary.

File: package.json (hunk 4 above)
Comment (praniredhat) [reply package.json]: webpack server never starts with inline mode for beggining, we have to check it. so leaving it as it was.

**Comment by praniredhat** [comment]
@vrathee shall i merge this to master

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,4 @@
-node_modules
+node_modules/*
```
Comment (vrathee) [inline .gitignore]: add yarn.lock file in the gitignore file and also remove if its present

File: app/cases/services/discussionService.js [hunk 7]
```diff
@@ -33,6 +33,9 @@ export default class DiscussionService {
                 attachPromise = AttachmentsService.getAttachments(caseId)
                     .then(() => this.updateElements());
             }
+            console.log("Updating Token Forcefully")
+            await window.sessionjs.updateToken(true);
+            console.log("Updating Token Finish")
```
Comment (vrathee) [inline app/cases/services/discussionService.js]: I thought we were going to do it in strata js if it works on QA before merging in master..

Comment (vrathee) [reply app/cases/services/discussionService.js]: Also Remove console logs

File: app/server.js (hunk 3 above)
Comment (vrathee) [reply app/server.js]: This change is still pending

File: bump_version.sh [hunk 8]
```diff
@@ -9,5 +9,5 @@ versiony --version=$1 --to=package.json,bower.json &&
...
-git push upstream master &&
+git push upstream s2i-ohc &&
```
Comment (vrathee) [inline bump_version.sh]: it will be master now

**Comment by praniredhat** [comment]
@vrathee Merging this PR 
This same code is going to Prod 
After this code is merged @anujsi  and @aksgupta14  will be able deploy their respective changes to QA for testing.

**Pull request merged**